src/Compilers/CSharp/Portable/CodeGen/EmitArrayInitializer.cs

```diff
@@ -469,5 +476,43 @@ private int TryGetRawDataForArrayInit(BoundArrayInitialization initializer, out
             data = writer.ToImmutableArray();
             return elementCount;
         }
+
+        /// <summary>
+        ///  Returns a byte blob that matches serialized content of zero-valued array.
+        /// </summary>
+        private int TryGetRawDataForDefaultArray(BoundArrayCreation initializer, out ImmutableArray<byte> data)
+        {
+            data = default;
+
+            if (initializer.Bounds.Length != 1)
+            {
+                return -1;
+            }
+
+            var elementCountValue = initializer.Bounds[0].ConstantValue;
+            if (elementCountValue == null)
+            {
+                return -1;
+            }
+
+            var elementCount = elementCountValue.Int32Value;
+            if (elementCount == 0)
+            {
+                data = ImmutableArray<byte>.Empty;
+                return 0;
+            }
+
+            var arrayType = (ArrayTypeSymbol)initializer.Type;
+            var elementType = arrayType.ElementType.SpecialType;
```

Comment:
Is there a better way to get the type?

Reply:
Should I do a safe cast or `initializer.Type` always will be an instance of `ArrayTypeSymbol` considering `initializer` is an instance of `BoundArrayCreation`?

Reply:
Getting the `SpecialType` seems wrong here. Could you test with an array of reference types, or enum types?

Reply:
I've refactored and reused the same value been retrieved at the line 371.